cli/tests/integrations/test_package.py

```diff
@@ -665,11 +669,37 @@ def test_list_cli():
 
     stdout = file_json(
         'tests/data/package/json/test_list_helloworld_cli.json')
-    _list(stdout=stdout)
+    _list(args=['--json'], stdout=stdout)
 
     _uninstall_cli_helloworld()
 
 
+def test_list_cli_flag():
+    kafka_path = 'tests/data/package/json/test_list_kafka_entry.json'
+    kafka_json = file_json_ast(kafka_path)
+    helloworld_path = 'tests/data/package/json/test_list_helloworld_entry.json'
+    helloworld_json = file_json_ast(helloworld_path)
+
+    def assert_test_case(args, expected_packages):
+        stdout = _list_stdout(args=['--json', '--cli'] + args)
+        actual_packages = json.loads(stdout.decode('utf-8'))
+        assert_same_elements(expected_packages, actual_packages)
```

Comment:
@tamarrow here's why I changed `_list()` to return `stdout` in my original PR: I needed to parse the JSON output. So I've created the `_list_stdout()` method to do that rather than change the behavior of `_list()`.